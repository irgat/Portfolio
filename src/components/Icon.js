import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    imagePath: PropTypes.string.isRequired,
    url: PropTypes.string,
    target: PropTypes.string,
};

const defaultProps = {
    url: '#',
    target: '_self',
};

const Icon = (props) => {
    return (
        <li>
            <a href={props.url} target={props.target}>
                <img src={props.imagePath} alt="" />
            </a>
        </li>
    );
};

Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;

export default Icon;
