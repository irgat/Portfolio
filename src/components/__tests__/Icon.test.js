import React from 'react';
import Enzyme from 'enzyme';
import Icon from '../Icon';

const data = [
    { imagePath: 'folderName/fileName.ext' },
    { imagePath: 'folderName/fileName.ext', url: 'https://www.w3.org' },
    { imagePath: 'folderName/fileName.ext', url: 'https://www.w3.org', target: '_blank' },
    { imagePath: 'folderName/fileName.ext', target: '_blank' },
];

const icon = (props) => {
    return Enzyme.shallow(<Icon {...props} />);
};

describe.each(data)('<Icon />', (props) => {
    it('renders a <li>', () => {
        expect(icon(props).find('li').getElements()).toMatchSnapshot();
    });

    describe('<li />', () => {
        it('renders an <a>', () => {
            expect(icon(props).find('a')).toHaveLength(1);
        });

        describe('<a />', () => {
            it('renders an <img>', () => {
                expect(icon(props).find('img')).toHaveLength(1);
            });
        });
    });
});
