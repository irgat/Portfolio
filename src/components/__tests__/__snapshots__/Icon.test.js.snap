// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders a <li> 1`] = `
Array [
  <li>
    <a
      href="#"
      target="_self"
    >
      <img
        alt=""
        src="folderName/fileName.ext"
      />
    </a>
  </li>,
]
`;

exports[`<Icon /> renders a <li> 2`] = `
Array [
  <li>
    <a
      href="https://www.w3.org"
      target="_self"
    >
      <img
        alt=""
        src="folderName/fileName.ext"
      />
    </a>
  </li>,
]
`;

exports[`<Icon /> renders a <li> 3`] = `
Array [
  <li>
    <a
      href="https://www.w3.org"
      target="_blank"
    >
      <img
        alt=""
        src="folderName/fileName.ext"
      />
    </a>
  </li>,
]
`;

exports[`<Icon /> renders a <li> 4`] = `
Array [
  <li>
    <a
      href="#"
      target="_blank"
    >
      <img
        alt=""
        src="folderName/fileName.ext"
      />
    </a>
  </li>,
]
`;
